src/logic/api-gateway.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+const get = require('lodash.get');
+
+const asApiGatewayResponse = (resp, stringifyJson = true) => {
+  if (get(resp, 'isApiResponse') !== true) {
+    throw resp;
+  }
+  if (get(resp, 'isApiError') === true) {
+    const body = {
+      message: resp.message,
+      messageId: resp.messageId,
+      context: resp.context
+    };
+    return {
+      statusCode: resp.statusCode,
+      body: stringifyJson ? JSON.stringify(body) : body
+    };
+  }
+  assert(get(resp, 'isApiError') === false);
+  const headers = resp.headers;
+  let body = resp.payload;
+  const isJsonResponse = get(resp, 'isJsonResponse') === true;
+  if (isJsonResponse && stringifyJson) {
+    body = JSON.stringify(body);
+  }
+  const isBinaryResponse = get(resp, 'isBinaryResponse') === true;
+  if (isBinaryResponse) {
+    body = body.toString('base64');
+  }
+  return {
+    statusCode: resp.statusCode,
+    body,
+    ...(Object.keys(headers).length === 0 ? {} : { headers }),
+    ...(isBinaryResponse ? { isBase64Encoded: true } : {})
+  };
+};
+module.exports.asApiGatewayResponse = asApiGatewayResponse;
+
+module.exports.wrap = ({
+  handler,
+  request,
+  route,
+  router,
+  module,
+  params = []
+}) => async (event, context) => {
+  if (!event.httpMethod) {
+    return Promise.resolve('OK - No API Gateway call detected.');
+  }
+  const kwargs = {
+    request,
+    event,
+    context,
+    route,
+    router,
+    params
+  };
+  const apply = [
+    async () => {
+      await module.before(kwargs);
+      kwargs.response = await handler(context.parsedParameters, context, event);
+    },
+    async () => module.after(kwargs)
+  ];
+  let isError = false;
```

Comment:
Should this be outside the `for` loop?

Reply:
The initial idea was that once an error is thrown it can't be converted into a non error. But that is not a case that can currently happen. So it really doesn't matter... I'll move this in and refactor it a bit more...

Reply:
So this actually matters. Because an error can be passed through the for loop. So the `isError` state needs to be passed through as well